File: Java/Easy/Accepted/find-the-middle-index-in-array_8.java
class Solution {
    public int findMiddleIndex(int[] nums) {
        
        int i = 0;
        int sum = 0;
        int leftSum = 0;
        for (int n : nums)
            sum += n;
        while (i < nums.length) {
            if (leftSum == sum - nums[i] - leftSum)
                return i;
            leftSum += nums[i];
            i++;
        }
        return -1;
    }
}

File: Rust/Easy/Accepted/find-the-middle-index-in-array_8.rs
impl Solution {
    pub fn find_middle_index(nums: Vec<i32>) -> i32 {
        
        let mut sum = nums.iter().sum::<i32>();
        let mut left_sum = 0;
        for (i, &num) in nums.iter().enumerate() {
            if left_sum == sum - left_sum - num {
                return i as i32;
            }
            left_sum += num;
        }
        -1
    }
}